toolchain/sem_ir/ids.h

```diff
@@ -281,17 +281,26 @@ struct RuntimeParamIndex : public IndexBase,
   // An explicitly invalid index.
   static const RuntimeParamIndex Invalid;
 
+  // An placeholder for index whose value is not yet known.
+  static const RuntimeParamIndex Unknown;
+
   using IndexBase::IndexBase;
 
   auto Print(llvm::raw_ostream& out) const -> void {
     out << "runtime_param";
-    IndexBase::Print(out);
+    if (*this == Unknown) {
+      out << "<unknown>";
+    } else {
+      IndexBase::Print(out);
+    }
   }
 };
 
 constexpr RuntimeParamIndex RuntimeParamIndex::Invalid =
     RuntimeParamIndex(InvalidIndex);
 
+constexpr RuntimeParamIndex RuntimeParamIndex::Unknown = RuntimeParamIndex(-2);
```

Comment:
Elsewhere we fairly consistently use `InvalidIndex - 1` for this kind of thing rather than assuming that `InvalidIndex` has a particular value. (I think we do end up assuming that in practice in various places but it seems good to be consistent.)
```suggestion
constexpr RuntimeParamIndex RuntimeParamIndex::Unknown =
    RuntimeParamIndex(InvalidIndex - 1);
```